### DISCUSSION TITLE
stocking upload

### DISCUSSION THREAD

**Discussion by DenuxPlays** [nextcloud/all-in-one#1768] [Questions] [answered]
Hey,
I have an issue with my upload speed.
I try to upload a large file (8.8 GB) to my Nextcloud and at the beginning my speed is normal (6 MB/s) but then it just pauses or hanging I don't know for like 2 seconds, and then it's start again. So my file needs 2 hours to upload instead of like half an hour or something like this.

Server configuration:
Virtualization: KVM
CPU: 4vCores
RAM: 8 GB (3-4 GB free)
Disk: 240 GB SSD
Network: 200Mbit/s

The performance don't seem to be the issue because nothing it at nearly 100%.
Does anyone know how to fix this?

**Comment by szaimen** [comment]
Hey, how are you trying to upload this file? Via Nextcloud's webinterface or via the Desktop client?

**Reply by DenuxPlays** [reply]
Tried it with both. Exact the same on both.

**Reply by szaimen** [reply]
Is clamav enabled?

**Reply by DenuxPlays** [reply]
No only office and imaginary are enabled.
Had an issue with clamAV before thats why I disabled it.

**Comment by szaimen** [comment]
Hi, so I tried to reproduce this with a 2.7GB big file on my test instance and I didn't succeed. It uploaded the file in expected 10min without any pausing or slow-downs. So this is likely something in your infrastracture that you need to fix.

BTW, are you using a Reverse Proxy?

**Reply by DenuxPlays** [reply]
Yes I am.
Apache (or httpd on my distro) and a non modified version of the example in the reverse-proxy.md file.
I calculated my average upload speed and it was like 1.3 MB/s when I am uploading it to my nextcloud. When I uploaded the exact same file to google drive I got an average of 7 MB/s.
On a mobile device the upload is not stocking its just very slow.

I am also using cloudflare but I disabled the proxy so that should not be the issue.

**Reply by DenuxPlays** [reply]
I am also using an .dev domain so http connections are not available.
That caused some issues before but not related to nextcloud.
Maybe that has some impact but I do not think so.

**Reply by szaimen** [reply]
All right! So when you now try to log in in an incognito tab on your Nextcloud with wrong credentials, which ip-address is logged in the server logs of your Nextcloud (under `https://yourdomain.com/settings/admin/logging`) ?

**Reply by DenuxPlays** [reply]
Remote IP: 109.40.241.xxx

**Reply by szaimen** [reply]
Is that the ip of your server or the one of the network where you are currently? 


If you add the ip-address to trusted_proxies in your config.php (e.g. `sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ config:system:set trusted_proxies 2 --value=<ip-address>
`) and reset the bruteforce protection via `sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ security:bruteforce:reset <ip-address>`, does this make things work?

**Reply by DenuxPlays** [reply]
No that is not the IP adress of my server.
I'll try it and give you an answer as soon as I am finished.

**Reply by szaimen** [reply]
> No that is not the IP adress of my server.

okay, then it will probably not help in your case.

**Reply by DenuxPlays** [reply]
Nope no changes sadly.
Do you have any other ideas?

**Reply by szaimen** [reply]
No idea currently, sorry. However you could have a look at this: https://github.com/nextcloud/all-in-one/discussions/1335

**Reply by DenuxPlays** [reply]
Okey.
Already looked at it.
I debug the whole thing later that day when I am at home.
If I find a solution I will share it.
Thank you for your help anyways 👍

**Reply by DenuxPlays** [reply]
> No idea currently, sorry. However you could have a look at this: #1335

I found the "error" if you can say this.
I had to disable http2 and now I am getting my normal upload speeds.
I do not know why http2 is slowing down my upload speed.
Here is the forum post I found: https://help.nextcloud.com/t/slow-upload-speed-i-need-advice/60909/14

**Comment by szaimen** [comment, marked as answer]
Ah I see. Great that you found the problem! Can you check if https://github.com/nextcloud/server/issues/25297#issuecomment-1281023661 resolves the issue for you as well? Then we can add it to our reverse proxy documentation

**Reply by DenuxPlays** [reply]
Yes this solves it.
I'll open a pr in a minute.

**Reply by szaimen** [reply]
> I'll open a pr in a minute.

Thanks a lot! :)

**Reply by szaimen** [reply]
fixed with https://github.com/nextcloud/all-in-one/pull/1776
